### PR TITLE
X-Admin app improvements

### PR DESCRIPTION
**Pull request by Velua**
- [x] Remove in house hash routing, adopt [React Router Hash Routing](https://reactrouter.com/en/main/router-components/hash-router).
- [x] Remove custom hooks for queries and adopt React Query. 
- [x] Zod type API responses / Centralised all chain API requests. 
- [x] Add new dashboard page alongside graphana
- [x] Remove old boot page / form
- [x] Add setup / Join wizard

### CODE REVIEW

**Comment by James-Mart** [comment]
Locally, I merged with the latest main, built and tested this. But I'm getting this ZodError on validating the `http_timeout` field of the config:
image

If I change the zod schema to expect a string then it works and lets me boot.

**Comment by James-Mart** [comment]
This looks very awesome

**Review by James-Mart — CHANGES_REQUESTED** [review]
I left various feedback on [this issue](https://github.com/gofractally/psibase/issues/742) that I gathered while using the interface. Nothing I'd block this PR over.

Only thing that seems necessary to fix before merge (from a usability perspective) is the [zod parse error](https://github.com/gofractally/psibase/pull/761#issuecomment-2269282618) that prevents the chain from booting.

**Comment by James-Mart** [comment]
Arrow button covered by expanded Advanced tab
image

**Comment by James-Mart** [comment]
My config seems to get corrupted at some point... a "service" key is getting deleted somehow
image

**Comment by James-Mart** [comment]
Also if I start out with `http_timeout = 4s`, I notice later in my config it is saying 
`http-timeout = 4000 s`

Is it converting to `ms` and using the wrong label?

**Comment by James-Mart** [comment]
If I add a peer, and then I click "disconnect", I think there's two things that are strange:
1. After a while, the connection is automatically re-established. This is unsurprising because presumably the peer is still in the configured list of peers, but the UX feels strange.
2. If I disconnect, and then click the "..." button, the option still says "Disconnect" instead of "Connect"
image

**Comment by swatanabe** [comment]
> Also if I start out with `http_timeout = 4s`, I notice later in my config it is saying `http-timeout = 4000 s`
> 
> Is it converting to `ms` and using the wrong label?

If that's happening, it's probably a bug on the native side. The `/native/admin/config` endpoint takes an integer in microseconds, with no units.

**Comment by Velua** [comment]
> If I add a peer, and then I click "disconnect", I think there's two things that are strange:
> 
>     1. After a while, the connection is automatically re-established. This is unsurprising because presumably the peer is still in the configured list of peers, but the UX feels strange. I think we can probably improve this UX later.
> 
>     2. If I disconnect, and then click the "..." button, the option still says "Disconnect" instead of "Connect"
>        image

Yeah I think the solution here is just to prevent crossing lanes, 
- Transient connections get disconnect
- Persistent connections get remove

It's awkward if persistent connections get both only to reconnect again anyway. 

- [ X ] Render "disconnect" button only on active transient connections. 
- [ X ] Render "remove" only on online & offline persistent connections.

**Comment by swatanabe** [comment]
> My config seems to get corrupted at some point... a "service" key is getting deleted somehow

It's a native bug that seems to be caused by having multiple entries for `localhost`

**Comment by James-Mart** [comment]
> Transient connections get disconnect

Although why even bother disconnecting if it's a transient connection, since it'll just come back again, right? Unless we can blacklist somehow @swatanabe

File: CMakeLists.txt
Comment (swatanabe) [inline CMakeLists.txt:707]: `psibase boot` needs to be updated

Comment (James-Mart) [reply CMakeLists.txt:739]: Tester as well, yeah?
https://github.com/gofractally/psibase/blob/98f3198a66f4754b29468ce783bd7a3496fd6219/rust/psibase/src/main.rs#L646
and
https://github.com/gofractally/psibase/blob/98f3198a66f4754b29468ce783bd7a3496fd6219/rust/psibase/src/tester.rs#L62

**Review by James-Mart — APPROVED** [review]

**Pull request merged**
